### DISCUSSION TITLE
SoftDelete and DB referential integrity

### DISCUSSION THREAD

**Discussion by blastiko** [wheels-dev/wheels#1276] [Q&A] [answered]
Hi guys
If I delete a record under softdelete manage, cfwheels rightly not delete record but write date in deletedat field.
Ok, this is not the problem, this work well.
The problem is when there are associative records in other tables: this records doesn't mark as delete, remain "valid".
Is there any automartic method to mark deleted (set date to deletedat field) in table's record associate when i delete a "master" record?
Eg. Authors table and Books table. Books table has a foreing key (authorid) field with CASCADE as a rule (in DB). When "master" record (author) is fisical deleted, all the books's record deleted too.
If i delete a record in Authors table using cfwheels softdelete this record doesn't really delete and so the book's record (of this author) doesn't mark as delete too.

**Comment by dbelanger** [comment, marked as answer]
Hey @blastiko - Quite right that the DB won't cascade the delete because that's for hard deletes done by the database.  Luckily, wheels has you covered with the "dependent" property in the [hasMany ](https://api.cfwheels.org/v2.4#model.hasmany)function.  By default it's false but if you set it to true, it will do what you want. 

Warning, if you have several levels of nested objects, you'll need to set dependent to true for every level to have your softDeletes carry through all the way to the lowest dependent object.

**Comment by blastiko** [comment]
Hi dbelanger
Thank you.
I had missed this "hasmany" function's parameter.
